src/main/java/online/shop/dao/jdbc/ProductDaoJdbsImpl.java

```diff
@@ -0,0 +1,111 @@
+package online.shop.dao.jdbc;
+
+import java.sql.ResultSet;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.PreparedStatement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import online.shop.dao.ProductDao;
+import online.shop.exception.DataProcessingException;
+import online.shop.lib.Dao;
+import online.shop.model.Product;
+import online.shop.util.ConnectionUtil;
+
+@Dao
+public class ProductDaoJdbsImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO product_table (name, price) VALUES (?, ?);";
+            PreparedStatement statement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                product.setId(resultSet.getLong(1));
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add item: " + product, e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM product_table WHERE product_id = ? AND deleted=false;";
+            PreparedStatement statement =
```

Comment:
```suggestion
            PreparedStatement statement = connection.prepareStatement(query);
```
Can we inline declaration and initialization here?